### ISSUE TITLE
Version number consistency

### ISSUE THREAD

**Issue by GunnarFarneback** [JuliaRegistries/RegistryCI.jl#432]
Nothing good will come if the registered version number does not match the version number in `Project.toml` in the given treehash. That these are consistent should be verified and should be easy to do as part of the "code can be loaded" check.

**Comment by ericphanson** [comment]
True, though we only run automerge on registrator PRs currently and registrator gets its version from the source’s Project.toml, so there’d have to be a registrator bug for this to hit General at this time. So at least it can’t just be user-error. Agreed it’s a good check to have though.

**Comment by DilumAluthge** [comment]
I think it's fine to add the check even if we assume that Registrator always do it correctly. We have some other checks (e.g. the list of files modified in a PR) that similarly should never fail on  Registrator PR, but we still check it just as a internal consistency check.

**Comment by GunnarFarneback** [comment]
In addition to possibly catching Registrator bugs this is a useful consistency check for people who run RegistryCI on their own registries, which may get PRs from other sources.

**Comment by DilumAluthge** [comment]
I think this was fixed by https://github.com/JuliaRegistries/RegistryCI.jl/pull/631, so I'll close this issue:

https://github.com/JuliaRegistries/RegistryCI.jl/blob/7aaf32b32d8c9a2a1c6027704736296e289923d8/AutoMerge/src/guidelines.jl#L337-L339

Let me know if I'm incorrect, and I'll re-open this issue.

**Comment by DilumAluthge** [comment]
Closing as fixed by https://github.com/JuliaRegistries/RegistryCI.jl/pull/631
